server/api/src/rs_get_resource_info_for_operation.cpp

```diff
@@ -0,0 +1,69 @@
+#include "irods/get_resource_info_for_operation.h"
+
+#include "irods/getRemoteZoneResc.h"
+#include "irods/irods_logger.hpp"
+#include "irods/irods_resource_backport.hpp"
+#include "irods/irods_resource_redirect.hpp"
+
+#include <nlohmann/json.hpp>
+#include <fmt/format.h>
+
+#include <array>
+#include <cstring> // for strdup
+#include <string>
+
+int rs_get_resource_info_for_operation(rsComm_t* _rsComm, dataObjInp_t* _dataObjInp, char** _out_info)
+{
+    using log_api = irods::experimental::log::api;
+    if (_rsComm == nullptr || _dataObjInp == nullptr || _out_info == nullptr) {
+        return SYS_INVALID_INPUT_PARAM;
```

Comment:
`USER__NULL_INPUT_ERR` or `SYS_NULL_INPUT` might communicate the error a little better. Thoughts?

Reply:
Hm, i admit to being out of my depth here.  There's this analogous [source line](https://github.com/irods/irods/blob/dc1ec1ace96e9878f51b4fdbe51dfc02405919c4/server/api/src/rs_check_auth_credentials.cpp#L21) from earlier this year where @korydraughn decided to use `SYS_INVALID_INPUT_PARAM`.

Reply:
`SYS_NULL_INPUT` is acceptable to me. I personally dislike things like `USER_*` because it can be misleading sometimes.

`SYS_NULL_INPUT` and `SYS_INVALID_INPUT_PARAM` are pretty generic even with the `SYS_` prefix. They also express the problem well.

Here are some questions:
- Q. How important is it for APIs to return the same error codes for similar error cases?
    - For example, I pass a null argument to API A and API B. API A returns `SYS_NULL_INPUT`. API B returns `SYS_INVALID_INPUT_PARAM`.
- Q. How much better is `SYS_NULL_INPUT` than `SYS_INVALID_INPUT_PARAM` for this case and vice versa?

Reply:
> Q. How important is it for APIs to return the same error codes for similar error cases? 

I personally don't see the consistency as important so much as communicating something to the caller in an understandable manner. If similar situations call for the same error code, so be it. Hopefully that makes sense...

> Q. How much better is SYS_NULL_INPUT than SYS_INVALID_INPUT_PARAM for this case and vice versa?

I just saw `SYS_INVALID_INPUT_PARAM` with no log or rError messages in several places throughout the function and thought "How would the caller know which input is invalid, or how it is invalid?" That's how I arrived at suggesting a different error code.

Reply:
That's fair.

I totally agree that the error cases need to be logged too.

Anyone else?

Reply:
The above seems logical.

Reply:
Done, so ... closing this conversation.